source/OptimaJet.DWKit.StarterApplication/Services/UserService.cs

```diff
@@ -11,14 +14,60 @@ namespace OptimaJet.DWKit.StarterApplication.Services
     public class UserService : EntityResourceService<User>
     {
         public UserRepository EntityRepository { get; }
-
+        public IJsonApiContext JsonApiContext { get; }
+        public IOrganizationContext OrganizationContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
 
         public UserService(
             IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
             IEntityRepository<User> entityRepository,
             ILoggerFactory loggerFactory) : base(jsonApiContext, entityRepository, loggerFactory)
         {
             this.EntityRepository = (UserRepository)entityRepository;
+            JsonApiContext = jsonApiContext;
+            OrganizationContext = organizationContext;
+            CurrentUserContext = currentUserContext;
+        }
+
+        public override async Task<IEnumerable<User>> GetAsync()
+        {
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist)
+            {
+                return Enumerable.Empty<User>().AsQueryable();
+            }
+            else
+            {
+                var query = JsonApiContext.QuerySet;
+                if (query == null)
+                {
+                    query = new JsonApiDotNetCore.Internal.Query.QuerySet();
+                    JsonApiContext.QuerySet = query;
+                }
+                var value = OrganizationContext.HasOrganization ? OrganizationContext.OrganizationId.ToString() : "";
+                query.Filters.Add(new JsonApiDotNetCore.Internal.Query.FilterQuery("organization-header", value, "="));
+                return await base.GetAsync();
+            }
+        }
+        public override async Task<User> GetAsync(int id)
+        {
+            var currentUser = EntityRepository.GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
+            if (currentUser.Id == id)
+            {
+                return await base.GetAsync(id);
+            }
+            var users = await GetAsync();
+            return users.SingleOrDefault(u => u.Id == id);
+        }
+        public override async Task<User> UpdateAsync(int id, User resource)
+        {
+            var user = await GetAsync(id);
+            if (user == null)
```

Comment:
would we want to return null when a user isn't found? or would we want to tell the current user that the user wasn't found? (404)

Reply:
Returning null at this point was resulting in a not found error.  I did go ahead and modify the code to send a json exception with a 404 code.  The unit tests did not require change.

Reply:
Not found is a correct error. -- what was wrong without the null? like, what was the response body content?